m2cgen/interpreters/c/interpreter.py

```diff
@@ -46,3 +60,38 @@ def interpret(self, expr):
     def interpret_vector_val(self, expr, **kwargs):
         self.with_vectors = True
         return super().interpret_vector_val(expr, **kwargs)
+
+    # Both methods supporting linear algebra do several things:
+    #
+    # 1. Call super method with extra parameters. Super method will return a
+    #    string with a call to the respective linear algebra function;
+    # 2. Add variable declaration where the result of the operation will be
+    #    stored;
+    # 3. Add code returned from super method to the result code;
+    # 4. Return name of the variable with current result.
+
+    def interpret_bin_vector_expr(self, expr, *args):
+        var_name = self._cg.get_var_name()
+
+        # Result: string like "addVectors(v1, v2, <size>, <var_name>)"
+        value = super().interpret_bin_vector_expr(
+            expr, expr.output_size, var_name)
+
+        self._cg.add_code_line("double {}[{}];".format(
```

Comment:
Don't we have `add_var_declaration` for this?

Reply:
Done